lib/api/protocol/window/switchTo.js

```diff
@@ -0,0 +1,91 @@
+const ProtocolAction = require('../_base-action.js');
+
+/**
+ * Change focus to another window.
+ *
+ * The window to change focus to must be specified by its server assigned window handle. To find out the window handle, use `window.getAllHandles()` command.
+ *
+ * @example
+ * module.exports = {
+ *  'switch to another window': function (browser) {
+ *     browser
+ *       .navigateTo('https://nightwatchjs.org/__e2e/window/')
+ *       .click('#openWindowBttn')
+ *       .waitUntil(function () {
+ *         // wait until window handle for the new window is available
+ *         return new Promise((resolve) => {
+ *           browser.window.getAllHandles(function (result) {
+ *             resolve(result.value.length === 2);
+ *           });
+ *         });
+ *       })
+ *       .perform(async function () {
+ *         const originalWindow = await browser.window.getHandle();
+ *         const allWindows = await browser.window.getAllHandles();
+ *
+ *         // loop through to find the new window handle
+ *         for (const windowHandle of allWindows) {
+ *           if (windowHandle !== originalWindow) {
+ *             await browser.window.switchTo(windowHandle);
+ *             break;
+ *           }
+ *         }
+ *
+ *         const currentWindow = await browser.window.getHandle();
+ *         browser.assert.notEqual(currentWindow, originalWindow);
+ *       });
+ *   },
+ *
+ *   'switch to another window with ES6 async/await': async function (browser) {
+ *     await browser.navigateTo('https://nightwatchjs.org/__e2e/window/');
+ *     await browser.click('#openWindowBttn');
+ *
+ *     // wait until window handle for the new window is available
+ *     await browser.waitUntil(async function () {
+ *       const windowHandles = await browser.window.getAllHandles();
+ *
+ *       return windowHandles.length === 2;
+ *     });
+ *
+ *     const originalWindow = await browser.window.getHandle();
+ *     const allWindows = await browser.window.getAllHandles();
+ *
+ *     // loop through available windows to find the new window handle
+ *     for (const windowHandle of allWindows) {
+ *       if (windowHandle !== originalWindow) {
+ *         await browser.window.switchTo(windowHandle);
+ *         break;
+ *       }
+ *     }
+ *
+ *     const currentWindow = await browser.window.getHandle();
+ *     await browser.assert.notEqual(currentWindow, originalWindow);
+ *   }
+ * }
+ *
+ * @syntax .window.switchTo(windowHandle, [callback])
+ * @method window.switchTo
+ * @param {string} windowHandle The server assigned window handle, should be one of the strings that was returned in a call to `.window.getAllHandles()`.
```

Comment:
it can also take window name. https://www.selenium.dev/selenium/docs/api/javascript/module/selenium-webdriver/lib/webdriver_exports_TargetLocator.html

Reply:
window name is kept there for legacy driver support, but its use was deprecated in W3C. See [this](https://github.com/SeleniumHQ/selenium/blob/796b2d40cdff5bc32b419ca7af1251e0d7adbf34/javascript/node/selenium-webdriver/lib/webdriver.js#L2457) and [this](https://www.w3.org/TR/webdriver/#switch-to-window).

And since we are keeping the older commands for backward compatibility, I don't think the new commands also need to be backwards compatible?